lib/avo/fields/select_field.rb

```diff
@@ -31,12 +31,16 @@ def options_for_select
       end
 
       def label
-        if display_value
+        if enum.present?
+          if display_value
+            options[value]
+          else
+            value
+          end
+        elsif display_value
           value
-        elsif enum.present?
-          options[value]
         else
-          options.invert[value]
+          options.invert.stringify_keys[value]
```

Comment:
This conditional is more complex than I wanted it to be, but it works in all scenarios (enum hash, enum array, options hash and all versions with `display_value` `true/false`).